CMakeLists.txt

```diff
@@ -0,0 +1,845 @@
+  set(CP2K_GPU_ARCH_NUMBER_K20X 35)
+  set(CP2K_GPU_ARCH_NUMBER_K40 35)
+  set(CP2K_GPU_ARCH_NUMBER_K80 37)
+  set(CP2K_GPU_ARCH_NUMBER_P100 60)
+  set(CP2K_GPU_ARCH_NUMBER_V100 70)
+  set(CP2K_GPU_ARCH_NUMBER_A100 80)
+  set(CP2K_GPU_ARCH_NUMBER_Mi50 gfx906)
+  set(CP2K_GPU_ARCH_NUMBER_Mi100 gfx908)
+  set(CP2K_GPU_ARCH_NUMBER_Mi200 gfx90a)
+  set(CP2K_GPU_ARCH_NUMBER_Mi250X gfx90a)
+endif()
+
+if(CP2K_USE_ACCEL MATCHES "CUDA")
+
+  if(NOT DEFINED CMAKE_CUDA_STANDARD)
+    set(CMAKE_CUDA_STANDARD 11)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+  endif()
+
+  # P100 is the default target. We need that in cray environment for some weird
+  # reason. And it does not harm to define it anyway
+  set(CMAKE_CUDA_ARCHITECTURES 60)
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler")
+  enable_language(CUDA)
+
+  find_package(CUDAToolkit REQUIRED)
+
+  list(FIND CP2K_SUPPORTED_CUDA_ARCHITECTURES ${CP2K_WITH_GPU}
+       CP2K_GPU_SUPPORTED)
+
+  if(CP2K_GPU_SUPPORTED EQUAL -1)
+    message(
+      FATAL_ERROR
+        "GPU architecture (${CP2K_WITH_GPU}) is not supported. Please choose from: ${CP2K_SUPPORTED_CUDA_ARCHITECTURES}"
+    )
+  endif()
+
+  # set cuda architecture number and compilation flags
+  add_compile_definitions(__OFFLOAD_CUDA
+                          $<$<COMPILE_LANGUAGE:CUDA>:__OFFLOAD_CUDA>)
+
+  # set cuda architecture number and compilation flags. Taken from dbcsr
+  set(CP2K_ACC_ARCH_NUMBER ${CP2K_GPU_ARCH_NUMBER_${CP2K_WITH_GPU}})
+  set(CMAKE_CUDA_ARCHITECTURES ${CP2K_ACC_ARCH_NUMBER})
+  set(CUDA_ARCHITECTURES ${CP2K_ACC_ARCH_NUMBER})
+
+  message(STATUS "GPU target architecture: " ${CP2K_WITH_GPU})
+  message(STATUS "GPU architecture number: " ${CP2K_ACC_ARCH_NUMBER})
+  message(STATUS "GPU profiling enabled: " ${CP2K_WITH_CUDA_PROFILING})
+
+  if(WITH_CUDA_PROFILING)
+    find_library(
+      CUDA_NVTOOLSEXT nvToolsExt
+      PATHS ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}
+      DOC "Building with CUDA profiling requires the nvToolsExt CUDA library"
+          REQUIRED)
+    message(STATUS "Found nvToolsExt: ${CUDA_NVTOOLSEXT}")
+  endif()
+
+  set(CP2K_USE_CUDA ON)
+  message(STATUS "-- CUDA compiler and libraries found")
+elseif(CP2K_USE_ACCEL MATCHES "HIP")
+  enable_language(HIP)
+  # Find hip
+  find_package(hip REQUIRED IMPORTED CONFIG)
+  find_package(hipfft REQUIRED IMPORTED CONFIG)
+  find_package(hipblas REQUIRED IMPORTED CONFIG)
+
+  add_compile_definitions(__OFFLOAD_HIP __HIP_PLATFORM_AMD__)
+  add_compile_definitions($<$<COMPILE_LANGUAGE:CUDA>:__OFFLOAD_HIP>)
+  set(CMAKE_HIP_ARCHITECTURES gfx801 gfx900 gfx90a)
+  if(NOT CMAKE_BUILD_TYPE)
+    set(HIP_RELEASE_OPTIONS "-O3 -g -DNDEBUG")
+  elseif(${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo")
+    set(HIP_RELEASE_OPTIONS "-O3 -g -DNDEBUG")
+  elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release")
+    set(HIP_RELEASE_OPTIONS "-O3 -DNDEBUG")
+  elseif(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+    set(HIP_RELEASE_OPTIONS "-O0 -g")
+  endif()
```

Comment:
These should probably be consistent with the available `CMAKE_BUILD_TYPES` defined starting on line 562?

Reply:
I need to review this in more details before marking it as resolved. the compilation options need cleanup